pkg/network/setup/configstate.go

```diff
@@ -45,40 +45,57 @@ func NewConfigState(cc cacheCreator, vmiUID string) ConfigState {
 //
 // The discovery step can be executed repeatedly with no limitation.
 // The configuration step is allowed to run only once. Any attempt to run it again will cause a critical error.
-func (c ConfigState) Run(podIfaceName string, discoverFunc func() error, configFunc func() error) error {
-	state, err := c.read(podIfaceName)
-	if err != nil {
-		return err
-	}
+func (c ConfigState) Run(nics []podNIC, discoverFunc func(*podNIC) error, configFunc func(*podNIC) error) error {
+	var pendingNICs []podNIC
+	for _, nic := range nics {
+		ifaceName := nic.podInterfaceName
+		state, err := c.read(ifaceName)
+		if err != nil {
+			return err
+		}
 
-	switch state {
-	case cache.PodIfaceNetworkPreparationStarted:
-		return neterrors.CreateCriticalNetworkError(
-			fmt.Errorf("pod interface %s network preparation cannot be restarted", podIfaceName),
-		)
-	case cache.PodIfaceNetworkPreparationFinished:
-		return nil
+		switch state {
```

Comment:
Is it possible to get here with one or more pod interfaces who's state is `PodIfaceNetworkPreparationFinished`?

Reply:
Yes, we can assume it can happen, but it is either all or none (see [this](https://github.com/kubevirt/kubevirt/pull/9055/commits/fbd2e4a0c22731519d8fe4a2730f061b107c567e#r1071280239) note).
Anyway, any such interfaces will not be processed from here on (see line 66).

Reply:
Thank you.